Compillation error in HPC

I was trying to compile spglib in a HPC cluster. I am getting an error related to `gtest-populate`
.
.
.
.
[ 11%] Creating directories for 'gtest-populate'
[ 22%] Performing download step (git clone) for 'gtest-populate'
-- Had to git clone more than once:
          3 times.
CMake Error at gtest-subbuild/gtest-populate-prefix/tmp/gtest-populate-gitclone.cmake:31 (message):
  Failed to clone repository: 'https://github.com/google/googletest'
gmake[2]: *** [CMakeFiles/gtest-populate.dir/build.make:102: gtest-populate-prefix/src/gtest-populate-stamp/gtest-populate-download] Error 1
gmake[1]: *** [CMakeFiles/Makefile2:83: CMakeFiles/gtest-populate.dir/all] Error 2
gmake: *** [Makefile:91: all] Error 2
.
.
.
.

Can someone comment? Thanks in advance.

--- a/.github/workflows/step_test.yaml
+++ b/.github/workflows/step_test.yaml
@@ -84,6 +84,52 @@ jobs:
         with:
           workflowPreset: "${{ matrix.toolchain }}-ci"
         continue-on-error: ${{ matrix.experimental && inputs.mask-experimental}}
+  cmake-version:
+    name: >
+      🔨 CMake ${{ matrix.cmake }}
+      ${{ matrix.experimental && '[🧪 Experimental]' || '' }}
+    runs-on: ubuntu-latest
+    continue-on-error: ${{ matrix.experimental || false }}
+    strategy:
+      fail-fast: false
+      matrix:
+        cmake: [ "3.20", "latest", "latestrc" ]
+        include:
+          - cmake: latestrc
+            experimental: true
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - name: Install dependencies
+        run: |
+          python3 -m pip install pip-tools
+          pip-compile --extra=test --strip-extras pyproject.toml ${{ matrix.pre && '--pre' }}
+          python3 -m pip install -r requirements.txt
+      - uses: lukka/get-cmake@latest
+        with:
+          cmakeVersion: ${{ matrix.cmake }}
+      - name: Run CMake workflow
+        uses: lukka/run-cmake@v10.3
+        with:
+          workflowPreset: gcc-ci
+        continue-on-error: ${{ matrix.experimental && inputs.mask-experimental}}
+        if: matrix.cmake != '3.20'
+      - name: Run CMake manually
+        # Need CMake 3.25 to properly use the presets
+        run: |
+          cmake -B ./build -G Ninja \
+            -DSPGLIB_WITH_Fortran=ON \
+            -DSPGLIB_WITH_TESTS=ON \
+            -DSPGLIB_WITH_Python=ON \
+            -DCMAKE_BUILD_TYPE=Release \
+            -DCMAKE_C_COMPILER=gcc \
+            -DCMAKE_C_COMPILER=g++ \
+            -DCMAKE_C_COMPILER=gfortran
+          cmake --build ./build
+          ctest --test-dir ./build -j $(nproc) --output-on-failure
+        continue-on-error: ${{ matrix.experimental && inputs.mask-experimental}}
+        if: matrix.cmake == '3.20'
   sanitizer:
     name: >
       🧹 ${{ matrix.sanitizer }} +

--- a/ChangeLog.md
+++ b/ChangeLog.md
@@ -44,6 +44,7 @@ GitHub release pages and in the git history.
 - [\[#425\]](https://github.com/spglib/spglib/pull/425) - Update to codecov v4
 - [\[#402\]](https://github.com/spglib/spglib/pull/402) - Bump artifact actions to v4
 - [\[#402\]](https://github.com/spglib/spglib/pull/402) - Build and inspect python sdist
+- [\[#431\]](https://github.com/spglib/spglib/pull/431) - Test CMake versions
 
 ## v2.3.1 (10 Feb. 2024)
 

--- a/fortran/CMakeLists.txt
+++ b/fortran/CMakeLists.txt
@@ -81,12 +81,24 @@ include(FetchContent)
 
 # Get Spglib if it's run as stand-alone project
 if (NOT Spglib_IS_TOP_LEVEL)
-    FetchContent_Declare(Spglib
-            GIT_REPOSITORY https://github.com/spglib/spglib
-            GIT_TAG develop
-            FIND_PACKAGE_ARGS CONFIG
-    )
-    list(APPEND external_libs Spglib)
+    if (CMAKE_VERSION VERSION_LESS 3.24)
+        # TODO: Use by default when minimum is CMake 3.24
+        find_package(Spglib CONFIG QUIET)
+        if (NOT Spglib_FOUND)
+            FetchContent_Declare(Spglib
+                    GIT_REPOSITORY https://github.com/spglib/spglib
+                    GIT_TAG develop
+            )
+            list(APPEND external_libs Spglib)
+        endif ()
+    else ()
+        FetchContent_Declare(Spglib
+                GIT_REPOSITORY https://github.com/spglib/spglib
+                GIT_TAG develop
+                FIND_PACKAGE_ARGS CONFIG
+        )
+        list(APPEND external_libs Spglib)
+    endif ()
 endif ()
 FetchContent_MakeAvailable(${external_libs})
 

--- a/fortran/CMakeLists.txt
+++ b/fortran/CMakeLists.txt
@@ -80,7 +80,7 @@ set(external_libs)
 include(FetchContent)
 
 # Get Spglib if it's run as stand-alone project
-if (NOT Spglib_IS_TOP_LEVEL)
+if (NOT Spglib_SOURCE_DIR)
     if (CMAKE_VERSION VERSION_LESS 3.24)
         # TODO: Use by default when minimum is CMake 3.24
         find_package(Spglib CONFIG QUIET)

--- a/python/CMakeLists.txt
+++ b/python/CMakeLists.txt
@@ -48,12 +48,24 @@ include(FetchContent)
 
 # Get Spglib if it's run as stand-alone project
 if (NOT Spglib_IS_TOP_LEVEL)
-    FetchContent_Declare(Spglib
-            GIT_REPOSITORY https://github.com/spglib/spglib
-            GIT_TAG develop
-            FIND_PACKAGE_ARGS CONFIG
-    )
-    list(APPEND external_libs Spglib)
+    if (CMAKE_VERSION VERSION_LESS 3.24)
+        # TODO: Use by default when minimum is CMake 3.24
+        find_package(Spglib CONFIG QUIET)
+        if (NOT Spglib_FOUND)
+            FetchContent_Declare(Spglib
+                    GIT_REPOSITORY https://github.com/spglib/spglib
+                    GIT_TAG develop
+            )
+            list(APPEND external_libs Spglib)
+        endif ()
+    else ()
+        FetchContent_Declare(Spglib
+                GIT_REPOSITORY https://github.com/spglib/spglib
+                GIT_TAG develop
+                FIND_PACKAGE_ARGS CONFIG
+        )
+        list(APPEND external_libs Spglib)
+    endif ()
 endif ()
 find_package(Python 3.8 COMPONENTS REQUIRED Interpreter Development.Module NumPy)
 FetchContent_MakeAvailable(${external_libs})

--- a/python/CMakeLists.txt
+++ b/python/CMakeLists.txt
@@ -47,7 +47,7 @@ set(external_libs)
 include(FetchContent)
 
 # Get Spglib if it's run as stand-alone project
-if (NOT Spglib_IS_TOP_LEVEL)
+if (NOT Spglib_SOURCE_DIR)
     if (CMAKE_VERSION VERSION_LESS 3.24)
         # TODO: Use by default when minimum is CMake 3.24
         find_package(Spglib CONFIG QUIET)

--- a/test/CMakeLists.txt
+++ b/test/CMakeLists.txt
@@ -73,14 +73,34 @@ endif ()
 set(external_libs)
 include(FetchContent)
 
-# Get Spglib if it's run as stand-alone project
-if (NOT Spglib_IS_TOP_LEVEL)
-    FetchContent_Declare(Spglib
-            GIT_REPOSITORY https://github.com/spglib/spglib
-            GIT_TAG develop
+set(fetchcontent_extra_args)
+if (CMAKE_VERSION VERSION_GREATER_EQUAL 3.24)
+    # TODO: Use by default when minimum is CMake 3.24
+    list(APPEND fetchcontent_extra_args
             FIND_PACKAGE_ARGS CONFIG
     )
-    list(APPEND external_libs Spglib)
+endif ()
+
+# Get Spglib if it's run as stand-alone project
+if (NOT Spglib_IS_TOP_LEVEL)
+    if (CMAKE_VERSION VERSION_LESS 3.24)
+        # TODO: Use by default when minimum is CMake 3.24
+        find_package(Spglib CONFIG QUIET)
+        if (NOT Spglib_FOUND)
+            FetchContent_Declare(Spglib
+                    GIT_REPOSITORY https://github.com/spglib/spglib
+                    GIT_TAG develop
+            )
+            list(APPEND external_libs Spglib)
+        endif ()
+    else ()
+        FetchContent_Declare(Spglib
+                GIT_REPOSITORY https://github.com/spglib/spglib
+                GIT_TAG develop
+                FIND_PACKAGE_ARGS CONFIG
+        )
+        list(APPEND external_libs Spglib)
+    endif ()
 endif ()
 
 set(BUILD_GMOCK OFF)
@@ -89,7 +109,8 @@ set(gtest_disable_pthreads ON)
 FetchContent_Declare(GTest
         GIT_REPOSITORY https://github.com/google/googletest
         GIT_TAG v1.13.0
-        FIND_PACKAGE_ARGS CONFIG
+        # For CMake < 3.24, always download GTest. Packaging for CMake  <3.24 is unsupported
+        ${fetchcontent_extra_args}
 )
 list(APPEND external_libs GTest)
 

--- a/test/CMakeLists.txt
+++ b/test/CMakeLists.txt
@@ -82,7 +82,7 @@ if (CMAKE_VERSION VERSION_GREATER_EQUAL 3.24)
 endif ()
 
 # Get Spglib if it's run as stand-alone project
-if (NOT Spglib_IS_TOP_LEVEL)
+if (NOT Spglib_SOURCE_DIR)
     if (CMAKE_VERSION VERSION_LESS 3.24)
         # TODO: Use by default when minimum is CMake 3.24
         find_package(Spglib CONFIG QUIET)

--- a/test/functional/fortran/CMakeLists.txt
+++ b/test/functional/fortran/CMakeLists.txt
@@ -49,6 +49,7 @@ create_test_sourcelist(FortranTests_Files
         test_fortran_version.F90
 )
 
+# CMake < 3.29 includes the driver source in the source list
 list(REMOVE_ITEM FortranTests_Files fortran_test.c)
 target_sources(Spglib_Fortran_tests PRIVATE
         ${CMAKE_CURRENT_BINARY_DIR}/fortran_test.c
@@ -65,6 +66,10 @@ target_compile_definitions(Spglib_Fortran_tests PRIVATE
 
 # Loop over all main tests files
 foreach (test IN LISTS FortranTests_Files)
+    if (test MATCHES ".*fortran_test.*")
+        # Skip the test driver
+        continue()
+    endif ()
     cmake_path(REMOVE_EXTENSION test OUTPUT_VARIABLE test_name)
     # Check if we have subtests defined
     if (DEFINED SubTests_${test_name})

--- a/test/functional/fortran/CMakeLists.txt
+++ b/test/functional/fortran/CMakeLists.txt
@@ -65,8 +65,7 @@ target_compile_definitions(Spglib_Fortran_tests PRIVATE
 
 # Loop over all main tests files
 foreach (test IN LISTS FortranTests_Files)
-    # TODO: CMake 3.20 Switch to cmake_path
-    get_filename_component(test_name ${test} NAME_WE)
+    cmake_path(REMOVE_EXTENSION test OUTPUT_VARIABLE test_name)
     # Check if we have subtests defined
     if (DEFINED SubTests_${test_name})
         # Loop over all subtests

